File: yuchen-dw-udfs/src/main/scala/com/yuchen/dw/udfs/scala/example/ExampleUDF.scala
package com.yuchen.dw.udfs.scala.example

import org.apache.hadoop.hive.ql.udf.generic.GenericUDF
import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspector

/**
 * @Author: xiaozhennan
 * @Date: 2022/11/22 11:26
 * @Package: com.yuchen.dw.udfs.scala.common
 * @ClassName: CommonUDF
 * @Description: $END
 * */
class ExampleUDF extends GenericUDF {
  override def initialize(objectInspectors: Array[ObjectInspector]): ObjectInspector = {
    objectInspectors(0)
  }

  override def evaluate(deferredObjects: Array[GenericUDF.DeferredObject]): AnyRef = {
    "test example udf"
  }

  override def getDisplayString(strings: Array[String]): String = {
    "test msg"
  }
}

File: yuchen-dw-udfs/src/main/java/com/yuchen/dw/udfs/java/common/ExampleUDF.java
package com.yuchen.dw.udfs.java.common;
import org.apache.hadoop.hive.ql.exec.UDF;


/**
 * 这种是过时写法,也可以用
 */
public class ExampleUDF   extends UDF{
    public  String evaluate(String str) {
        try {
            return "HelloWorld " + str;
        } catch (Exception e) {
            return null;
        }
    }
}